Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -341,32 +453,263 @@ define([
+                return true;
+            }
+        });
+        return defined(result);
+    }
+
+    function getPropertyNamesInHierarchy(batchTable, batchId, names) {
+        var hierarchy = batchTable._batchTableHierarchy;
+        traverseHierarchy(hierarchy, batchId, function(hierarchy, instanceIndex) {
+            var classId = hierarchy.classIds[instanceIndex];
+            var instances = hierarchy.classes[classId].instances;
+            for (var name in instances) {
+                if (instances.hasOwnProperty(name)) {
+                    names[name] = true;
+                }
+            }
+        });
+    }
+
+    function getHierarchyProperty(batchTable, batchId, name) {
+        var hierarchy = batchTable._batchTableHierarchy;
+        return traverseHierarchy(hierarchy, batchId, function(hierarchy, instanceIndex) {
+            var classId = hierarchy.classIds[instanceIndex];
+            var instanceClass = hierarchy.classes[classId];
+            var indexInClass = hierarchy.classIndexes[instanceIndex];
+            var propertyValues = instanceClass.instances[name];
+            if (defined(propertyValues)) {
+                if (defined(propertyValues.typedArray)) {
+                    return getBinaryProperty(propertyValues, indexInClass);
+                } else {
+                    return clone(propertyValues[indexInClass], true);
+                }
+            }
+        });
+    }
+
+    function setHierarchyProperty(batchTable, batchId, name, value) {
+        var hierarchy = batchTable._batchTableHierarchy;
+        var result = traverseHierarchy(hierarchy, batchId, function(hierarchy, instanceIndex) {
+            var classId = hierarchy.classIds[instanceIndex];
+            var instanceClass = hierarchy.classes[classId];
+            var indexInClass = hierarchy.classIndexes[instanceIndex];
+            var propertyValues = instanceClass.instances[name];
+            if (defined(propertyValues)) {
+                if (defined(propertyValues.typedArray)) {
+                    setBinaryProperty(propertyValues, indexInClass, value);
+                } else {
+                    propertyValues[indexInClass] = clone(value, true);
+                }
+                return true;
+            }
+        });
+        return defined(result);
+    }
+
+    Cesium3DTileBatchTable.prototype.isClass = function(batchId, className) {
+        var featuresLength = this.featuresLength;
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(batchId) || (batchId < 0) || (batchId > featuresLength)) {
+            throw new DeveloperError('batchId is required and between zero and featuresLength - 1 (' + featuresLength - + ').');
+        }
+        if (!defined(className)) {
+            throw new DeveloperError('className is required.');
+        }
+        //>>includeEnd('debug');
+
+        // PERFORMANCE_IDEA : cache results in the ancestor classes to speed up this check if this area becomes a hotspot
```

Comment:
The other `PERFORMANCE_IDEA` I think I mentioned is to replace the string compare with integers (implicit to the end user).